===============================================================================

This is a patch for Open vSwitch.

This patch fixes a compilation error occurring when building Open vSwitch for
the x86_64 target architecture. The error is due to the usage of the
csum_ipv6_magic in file stt.c without a proper #include directive.

It also fixes another error due to depmod being invoked to reconstruct module
dependencies for the kernel running on the host system (instead of the UML
kernel).

Applies successfully to commit bc651c85497d7a8c32e83b44986dae27dec3bcd5

===============================================================================

     Copyright (C) 2015  Massimo Rimondini (rimondini@ing.uniroma3.it)
     Computer Networks Research Group, Roma Tre University

     This program is free software: you can redistribute it and/or modify
     it under the terms of the GNU General Public License as published by
     the Free Software Foundation, version 3 of the License.

     This program is distributed in the hope that it will be useful,
     but WITHOUT ANY WARRANTY; without even the implied warranty of
     MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
     GNU General Public License for more details.

     You should have received a copy of the GNU General Public License
     along with this program.  If not, see <http://www.gnu.org/licenses/>.

=================================================================================
diff -Naurx .git ovs/datapath/linux/compat/stt.c ovs-patched/datapath/linux/compat/stt.c
--- ovs/datapath/linux/compat/stt.c	2015-07-02 10:34:27.000000000 +0200
+++ ovs-patched/datapath/linux/compat/stt.c	2015-07-02 10:36:00.000000000 +0200
@@ -27,6 +27,11 @@
 #include <linux/tcp.h>
 #include <linux/workqueue.h>
 
+/* The following include is required because of the usage of
+function csum_ipv6_magic.
+Omitting it causes errors on x86_64 builds. */
+#include <net/ip6_checksum.h>
+
 #include <net/icmp.h>
 #include <net/inet_ecn.h>
 #include <net/ip.h>
diff -Naurx .git ovs/datapath/linux/Makefile.main.in ovs-patched/datapath/linux/Makefile.main.in
--- ovs/datapath/linux/Makefile.main.in	2015-07-02 10:34:27.000000000 +0200
+++ ovs-patched/datapath/linux/Makefile.main.in	2015-07-02 10:36:26.000000000 +0200
@@ -72,7 +72,7 @@
 
 modules_install:
 	$(MAKE) -C $(KSRC) M=$(builddir) modules_install
-	depmod `sed -n 's/#define UTS_RELEASE "\([^"]*\)"/\1/p' $(KSRC)/include/generated/utsrelease.h`
+	depmod -b $(KERNEL_TARGET_MODULES_DIR) `sed -n 's/#define UTS_RELEASE "\([^"]*\)"/\1/p' $(KSRC)/include/generated/utsrelease.h`
 endif
 
 # Much of the kernel build system in this file is derived from Intel's
